### PR TITLE
Improvements: design support library

### LINKED ISSUE #361: Use new NavigationView
Update the left nav drawer with the `NavigationView` from design support library.
Navigation drawer menu should have sections as required by #336

### PR DESCRIPTION
**Pull request by PareshMayani**
Have made below changes:
- Integrated Navigation view and removed older stuffs
- Used TextInputLayout instead of FloatLabelLayout
- Replaced crouton with Snackbar 

Closes #361 #364 and #365

### CODE REVIEW

**Comment by tasomaniac** [comment]
@PareshMayani 
Can you include screenshots for the new Snackbars and NavigationView. 
It would be nicer if you can open separate Pull Requests for separate issues although they are related.

File: app/src/main/java/org/gdg/frisbee/android/chapter/NewsFragment.java [hunk 1]
```diff
@@ -149,8 +150,9 @@ public void onGet(Object item) {
                     ActivityFeed feed = (ActivityFeed) item;
 
                     if (isAdded()) {
-                        Crouton.makeText(getActivity(), R.string.cached_content,
```
Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/chapter/NewsFragment.java]: Above in `MainActivity` you created `mContentFrameLayout` and put the Snackbar into it. Why it is different here? Thank you.

Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/chapter/NewsFragment.java]: Oh I see. It is because Crouton's uses `Activity`'s `View` by default. But here `Snackbar` can insert itself into `Fragment`'s `View`

File: app/src/main/java/org/gdg/frisbee/android/common/GdgActivity.java [hunk 2]
```diff
@@ -233,7 +229,6 @@ private void resolveSignInError() {
     @Override
     public void onDestroy() {
         super.onDestroy();
-        Crouton.cancelAllCroutons();
```
Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/common/GdgActivity.java]: Do we require the same method for Snackbar implementation?

File: app/src/main/java/org/gdg/frisbee/android/common/GdgNavDrawerActivity.java [hunk 3]
```diff
@@ -121,8 +118,12 @@ public void onDrawerOpened(View drawerView) {
         };
         mDrawerLayout.setDrawerListener(mDrawerToggle);
 
+        if (mNavigationView != null) {
+            setupDrawerContent(mNavigationView);
+        }
     }
 
+/*
     @OnItemClick(R.id.navdrawer_list)
```
Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/common/GdgNavDrawerActivity.java]: No commenting out please. We can remove these lines if they are not needed.

File: app/src/main/java/org/gdg/frisbee/android/common/GdgNavDrawerActivity.java [hunk 4]
```diff
@@ -121,8 +118,12 @@ public void onDrawerOpened(View drawerView) {
...
     public void onDrawerItemClick(AdapterView<?> adapterView, View view, int i, long l) {
```
Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/common/GdgNavDrawerActivity.java]: And these lines should be implemented as well. 

```
if (PrefUtils.shouldOpenDrawerOnStart(GdgNavDrawerActivity.this)) {
             PrefUtils.setShouldNotOpenDrawerOnStart(GdgNavDrawerActivity.this);
         }

```

File: app/src/main/res/layout/activity_first_start.xml [hunk 5]
```diff
@@ -19,7 +19,8 @@
 <FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
   android:layout_width="match_parent"
   android:layout_height="match_parent"
-  android:orientation="vertical">
+  android:orientation="vertical"
+    android:id="@+id/contentLayout">
```
Comment (tasomaniac) [inline app/src/main/res/layout/activity_first_start.xml]: Indentation problem here.

File: app/src/main/res/layout/dialog_feedback.xml [hunk 6]
```diff
@@ -1,48 +1,44 @@
...
-      android:maxLines="4"
-      android:minLines="1"/>
+    <android.support.design.widget.TextInputLayout
+        android:id="@+id/feedback_message_text_layout"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content">
 
-  </org.gdg.frisbee.android.widget.FloatLabelLayout>
+        <EditText
+            android:id="@+id/feedback_message_text"
```
Comment (tasomaniac) [inline app/src/main/res/layout/dialog_feedback.xml]: Indentation problem here too. We use 2 space indentation in XML files.

File: app/src/main/res/layout/nav_header.xml [hunk 7]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:orientation="vertical"
+    android:layout_width="@dimen/navdrawer_width"
+    android:layout_height="@dimen/navdrawer_image_height">
+
+        <ImageView
```
Comment (tasomaniac) [inline app/src/main/res/layout/nav_header.xml]: Indentation problem here as well.

File: app/src/main/res/menu/drawer_view.xml [hunk 8]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <group android:checkableBehavior="single" android:id="@+id/navigation_menu">
+
+        <item
```
Comment (tasomaniac) [inline app/src/main/res/menu/drawer_view.xml]: Is this XML file used? I couldn't see it anywhere. I think the menu is setup dynamically.

File: app/src/main/res/layout/dialog_feedback.xml [hunk 9]
```diff
@@ -1,48 +1,44 @@
...
-  android:layout_height="match_parent"
-  android:orientation="vertical"
-  android:paddingLeft="20dp"
-  android:paddingRight="20dp">
-
-  <org.gdg.frisbee.android.widget.FloatLabelLayout
+    xmlns:app="http://schemas.android.com/apk/res-auto"
     android:layout_width="match_parent"
-    android:layout_height="wrap_content"
-    app:floatLabelTextAppearance="@style/TextAppearance.GDG.FloatLabel"
```
Comment (tasomaniac) [inline app/src/main/res/layout/dialog_feedback.xml]: I don't see the style for old FloatLabelLayout is preserved. What happened to 

```
    app:floatLabelTextAppearance="@style/TextAppearance.GDG.FloatLabel"
    app:floatLabelPaddingTop="8dp"
    app:floatLabelPaddingLeft="4dp"
    app:floatLabelPaddingBottom="0dp">
```

File: app/src/main/res/menu/drawer_view.xml (hunk 8 above)
Comment (PareshMayani) [inline app/src/main/res/menu/drawer_view.xml]: yes it's not needed now as we are adding drawer items pragmatically.

**Comment by PareshMayani** [comment]
@tasomaniac about "It would be nicer if you can open separate Pull Requests for separate issues although they are related."

Please let me know what should I do? I really don't know how can I easily breakdown this single pull requests into the multiple? If you know any easier way then please share!

**Comment by tasomaniac** [comment]
@PareshMayani Actually it is not that crucial. What you can do is that you can create a new branch with the issue name for every issue separately. And you can work on every single issue on separate brances. And then you can create multiple Pull Requests.

File: app/src/main/res/menu/drawer_view.xml (hunk 8 above)
Comment (tasomaniac) [inline app/src/main/res/menu/drawer_view.xml]: Can you delete it?

**Comment by PareshMayani** [comment]
@tasomaniac bro that I know :-) Anyway will manage it out and you will get separate pull requests!

**Comment by tasomaniac** [comment]
I think it is really fine for this PR :smile_cat: 
Can you look at the other comments I wrote?

File: app/src/main/res/layout/dialog_feedback.xml (hunk 6 above)
Comment (PareshMayani) [inline app/src/main/res/layout/dialog_feedback.xml]: @tasomaniac could you please make me updated with the 2 space indentation rule? How can I make sure the layout is as per the 2 space indentation rule? Because so far I have used CTRL+ALT+L for formatting in Android studio.

Comment (tasomaniac) [inline app/src/main/res/layout/dialog_feedback.xml]: In Android Studio project properties there is an option to specify the tab size and indent. You can set it only for this project. Then CTRL+ALT+L will format the XML code to 2 space indentation. 

When you have a 2space indented file opened, it also suggests you to change the setting.

screen shot 2015-07-19 at 14 54 46

Comment (PareshMayani) [inline app/src/main/res/layout/dialog_feedback.xml]: @tasomaniac thanks :) Have configured the options.

**Comment by Splaktar** [comment]
We should add an .editorconfig file for the project.

File: app/src/main/java/org/gdg/frisbee/android/common/GdgActivity.java (hunk 2 above)
Comment (PareshMayani) [inline app/src/main/java/org/gdg/frisbee/android/common/GdgActivity.java]: @tasomaniac there is any such method in Snackbar class!

Comment (tasomaniac) [inline app/src/main/java/org/gdg/frisbee/android/common/GdgActivity.java]: We may not need one. Forget about it :smile:

**Comment by PareshMayani** [comment]
@Splaktar indeed that would be helpful to the new contributor!

File: app/src/main/res/values/games-ids.xml [hunk 10]
```diff
@@ -6,11 +6,17 @@ Save this file as res/values/games-ids.xml in your project.
 <resources>
     <string name="app_id" translatable="false">429371117063</string>
     <string name="achievement_signin_palooza" translatable="false">CgkIh5yNxL8MEAIQAA</string>
-    <string name="achievement_cinephile" translatable="false">CgkIh5yNxL8MEAIQAQ</string>
     <string name="achievement_gdg_maniac" translatable="false">CgkIh5yNxL8MEAIQAg</string>
     <string name="achievement_thanks_for_coming_back." translatable="false">CgkIh5yNxL8MEAIQAw</string>
     <string name="achievement_king_of_the_hill" translatable="false">CgkIh5yNxL8MEAIQBA</string>
+    <string name="achievement_looking_for_experts" translatable="false">CgkIh5yNxL8MEAIQDw</string>
```
Comment (PareshMayani) [inline app/src/main/res/values/games-ids.xml]: @tasomaniac your contribution in my PR? I think it should be in separate PR as it's not related to design support library changes!

Comment (tasomaniac) [inline app/src/main/res/values/games-ids.xml]: Hey man :smile: the PR is originated from `PareshMayani:develop` which I don't have access. 
You somehow pulled my commit from branch `refactor/358-achievements` into your develop branch. You should remove it.
